### PR TITLE
Prevent ISE duplicate key when checking if instrumentation can be used

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/dev/ClassComparisonUtil.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ static boolean compareAnnotations(Collection<AnnotationInstance> a, Collection<A
             return false;
         }
         Map<DotName, AnnotationInstance> lookup = b.stream()
-                .collect(Collectors.toMap(AnnotationInstance::name, Function.identity()));
+                .collect(Collectors.toMap(AnnotationInstance::name, Function.identity(), (i1, i2) -> i1));
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/dev/ClassComparisonUtil.java:103]: Do you really want to ignore the second one?

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a question that (I think) needs an answer before merging.

**Comment by stuartwdouglas** [comment]
What actually causes this? There should not be a situation where there are duplicate annotations.

**Comment by gsmet** [comment]
Is it with latest Quarkus? Because we had a bug in method comparisons triggering exactly this. Martin fixed it and I believe it’s in 1.11.1.Final.

**Comment by evacchi** [comment]
I had this issue with one of the examples with Quarkus 1.11.x IIRC -- sorry couldn't come up with a stand-alone reproducer yet. Had two separate issues, this one with

- https://github.com/kiegroup/kogito-examples/tree/stable/process-scripts-quarkus -- while updating the BPMN file
- the other (different nature, still related to instrumentation) https://github.com/kiegroup/kogito-examples/tree/stable/dmn-quarkus-example -- while updating the DMN file

**Comment by gsmet** [comment]
Could someone have a look at this? I won't have the time to check it and PRs are piling up. Looks like something we would want fixed for 1.11.2.Final.

**Comment by geoand** [comment]
I'll try tomorrow but I can make no guarantees as there is a ton of (high priority) work pilling up

**Comment by geoand** [comment]
In any case, https://github.com/quarkusio/quarkus/pull/14775 provides a way to disable instrumentation altogether so we should definitely get that one in

**Comment by evacchi** [comment]
yesterday I could not reproduce this on our master, so no need to rush it. Moreover, the flag to disable instrumentation should be helpful as well.

**Comment by geoand** [comment]
@evacchi have you had any more troubles with this?

If not, I can close this

**Comment by evacchi** [comment]
let's close it for now @geoand

**Pull request closed without merging**